### PR TITLE
Adding a managed coherence test

### PR DESCRIPTION
**Pull request by bhavaniravichandran**

### CODE REVIEW

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItManagedCoherence.java [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+            };
+            verifyServersStatus(domain, pods, DOMAINUID);
+
+            // Build WAR in the admin pod and deploy it from the admin pod to a weblogic target
+            TestUtils.buildDeployCoherenceAppInPod(domain,
+                    testAppName, scriptName, BaseTest.getUsername(), BaseTest.getPassword(),
+                    appToDeploy, "dataCluster");
+
+            // Wait some time for deployment to get into Active state
+            Thread.sleep(60 * 1000);
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/ItManagedCoherence.java]: hard sleep for 60 seconds is too long. can u check for the status of app in loop or reduce the time if it doesn't need such long sleep?

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItManagedCoherence.java [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+            };
+            verifyServersStatus(domain, pods, DOMAINUID1);
+
+            // Build WAR in the admin pod and deploy it from the admin pod to a weblogic target
+            TestUtils.buildDeployCoherenceAppInPod(domain,
+                    testAppName, scriptName, BaseTest.getUsername(), BaseTest.getPassword(),
+                    appToDeploy, "dataCluster");
+
+            // Wait some time for deployment to get into Active state
+            Thread.sleep(60 * 1000);
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/ItManagedCoherence.java]: same as other comment

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItManagedCoherence.java [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+        String domain1LabelSelector = String.format("weblogic.domainUID in (%s)", domainUID);
+        String namespace = domain.getDomainNs();
+        for (String pod : pods) {
+            assertTrue(
+                    pod + " Pod not running", testUtil.isPodRunning(namespace, domain1LabelSelector, pod));
+        }
+    }
+
+    private void coherenceCacheTest() throws Exception {
+        //Assume.assumeFalse(QUICKTEST);
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/ItManagedCoherence.java]: remove commented code

File: integration-tests/src/test/java/oracle/kubernetes/operator/utils/TestUtils.java [hunk 4]
```diff
@@ -1664,6 +1664,105 @@ public static void buildDeployWebServiceAppInPod(
...
+            + " to respective clusters ");
+
+    // Create app dir in the admin pod
+    StringBuffer mkdirCmd = new StringBuffer(" -- bash -c 'mkdir -p ");
+    mkdirCmd.append(appLocationInPod + "'");
+
+    // Create app dir in the admin pod
+    kubectlexec(adminServerPod, domainNS, mkdirCmd.toString());
+
+/*
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/utils/TestUtils.java]: remove commented code

File: integration-tests/src/test/resources/apps/coherenceapp/gar/pof/Contact.java [hunk 5]
```diff
@@ -0,0 +1,129 @@
+<!-- Copyright 2019, Oracle Corporation and/or its affiliates.  All rights reserved.
```
Comment (vanajamukkara) [inline integration-tests/src/test/resources/apps/coherenceapp/gar/pof/Contact.java]: javadoc and comment format is not correct in these files? any reason?
should be 
// Copyright 2019, Oracle Corporation and/or its affiliates.  All rights reserved.
// Licensed under the Universal Permissive License v 1.0 as shown at
// http://oss.oracle.com/licenses/upl.

File: integration-tests/src/test/resources/apps/coherenceapp/gar/pof/Contact.java [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+import com.tangosol.io.pof.PofWriter;
+import com.tangosol.io.pof.PortableObject;
+
+import com.tangosol.util.Base;
+
+import java.io.IOException;
+
+import java.util.Date;
+
+/**
```
Comment (vanajamukkara) [inline integration-tests/src/test/resources/apps/coherenceapp/gar/pof/Contact.java]: fix javadoc format in the entire class
author can be removed

File: integration-tests/src/test/resources/apps/coherenceapp/gar/pof/ContactId.java [hunk 7]
```diff
@@ -0,0 +1,119 @@
+<!-- Copyright 2019, Oracle Corporation and/or its affiliates.  All rights reserved.
```
Comment (vanajamukkara) [inline integration-tests/src/test/resources/apps/coherenceapp/gar/pof/ContactId.java]: same comments as above for javadoc and comments format

File: integration-tests/src/test/resources/apps/coherenceapp/ear/CoherenceApp.java [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+import com.tangosol.net.NamedCache;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ *
+ * This web application is used for following with respect to Coherence cache depending on user action. 
+ * 1. Populating the cahce using "add" , first name and second name used for populating the data.
```
Comment (pfmackin) [inline integration-tests/src/test/resources/apps/coherenceapp/ear/CoherenceApp.java]: cahce spelling

File: integration-tests/src/test/resources/apps/coherenceapp/ear/CoherenceApp.java [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+		response.setContentType("text/html;charset=UTF-8");
+		PrintWriter out = response.getWriter();
+
+		String action = request.getParameter("action");
+		String firstName = request.getParameter("first");
+		String secondName = request.getParameter("second");
+
+		// Call respective methods based on user action
+		if (action.equals("add")) {
+			add(request, response);
```
Comment (pfmackin) [inline integration-tests/src/test/resources/apps/coherenceapp/ear/CoherenceApp.java]: normally it is get/put not get/add for access the cache

File: integration-tests/src/test/resources/apps/coherenceapp/ear/CoherenceApp.java [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+		NamedCache cache = CacheFactory.getCache("contacts");
+		String firstName = request.getParameter("first");
+		String secondName = request.getParameter("second");
+		PrintWriter out = response.getWriter();
+		Contact contact = new Contact(firstName, secondName);
+
+		ContactId contactID = new ContactId(firstName, secondName);
+		keyList.add(contactID);
+		cache.put(contactID, contact);
+		Contact contactGet = (Contact) cache.get(contactID);
```
Comment (pfmackin) [inline integration-tests/src/test/resources/apps/coherenceapp/ear/CoherenceApp.java]: why are you getting the data here?

File: integration-tests/src/test/resources/apps/coherenceapp/ear/CoherenceApp.java [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+		out.println("\nContact added:" + contactGet);
+	}
+
+	/*
+	 * Retrieve all cached data as of now
+	 */
+	private void get(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
```
Comment (pfmackin) [inline integration-tests/src/test/resources/apps/coherenceapp/ear/CoherenceApp.java]: this should be getAll. Get is just for a single cache entry

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItManagedCoherence.java [hunk 12]
```diff
@@ -0,0 +1,338 @@
...
+        //check if cache size is 6
+        result = getCacheSize();
+        logger.info("number of records in cache = " + result.stdout());
+        if (!(result.stdout().equals("6"))) {
+            logger.info("number of records in cache = " + result.stdout());
+            assertFalse("Expected 6 records", "6".equals(result.stdout()));
+        }
+        //get the data from cache
+        result = getCacheContents();
+        logger.info("Cache contains the following entries \n" + result.stdout());
```
Comment (pfmackin) [inline integration-tests/src/test/java/oracle/kubernetes/operator/ItManagedCoherence.java]: You should validate the cache contents.  Ideally this should be done in the app running in WebLogic, there is really no reason to transfer everything back to the client.

File: integration-tests/src/test/java/oracle/kubernetes/operator/utils/TestUtils.java [hunk 13]
```diff
@@ -1664,6 +1664,67 @@ public static void buildDeployWebServiceAppInPod(
...
+    final String appLocationInPod = BaseTest.getAppLocationInPod() + "/" + appName;
+    final String scriptPathOnHost = BaseTest.getAppLocationOnHost() + "/" + scriptName;
+    final String scriptPathInPod = BaseTest.getAppLocationInPod() + "/" + scriptName;
+
+    // Default values to build archive file
+    final String initInfoDirName = "WEB-INF";
+    String archiveExt = "war";
+    String infoDirName = initInfoDirName;
+    String domainNS = domain.getDomainNs();
+    String deployTargetForGAR = (args.length == 0) ? "dataCluster" : args[0];
```
Comment (pfmackin) [inline integration-tests/src/test/java/oracle/kubernetes/operator/utils/TestUtils.java]: It would be better if the targets were passed in as params

**Comment by pfmackin** [comment]
I had a few comments, nothing blocking.  In general this is a good start.  It would be nice to have another related test to load thousands of entries and validate the contents after scale-in / scale-out.  This logic should run in the Admin VM and not transfer the data back to the test.

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItManagedCoherence.java [hunk 14]
```diff
@@ -0,0 +1,338 @@
...
+
+        boolean testCompletedSuccessfully = false;
+        domain = null;
+
+        // create operator1
+        if (operator1 == null) {
+            operator1 = TestUtils.createOperator(OPERATOR1_YAML);
+        }
+
+        try {
```
Comment (pfmackin) [inline integration-tests/src/test/java/oracle/kubernetes/operator/ItManagedCoherence.java]: looks like alot of common code with testCreateCoherenceDomainOnPvUsingWlst could be shared

**Comment by pfmackin** [comment]
LGTM

**Comment by rjeberhard** [comment]
@pfmackin, you put LGTM but your comments weren't addressed; even the "easy" one like fixing the spelling of "cache".  Obviously, we should fix the spelling error, but what about the rest of your comments?

**Pull request merged**